sequence_processing_pipeline/NuQCJob.py

```diff
@@ -0,0 +1,288 @@
+                         f"OUTPUT={self.output_path}",
+                         f"TMPDIR={self.temp_dir}"]
+
+        job_params = ['-J', self.batch_prefix, '--array', batch_count,
+                      '--mem', '20G', '--export', ','.join(export_params)]
+
+        # job_script_path formerly known as:
+        #  process.multiprep.pangenome.adapter-filter.pe.sbatch
+        job_info = self.submit_job(job_script_path,
+                                   # job_parameters - was None
+                                   ' '.join(job_params),
+                                   # script_parameters
+                                   None,
+                                   # assume we want to exec from the log_path
+                                   # for now.
+                                   exec_from=self.log_path,
+                                   callback=callback)
+
+        job_id = job_info['job_id']
+        logging.debug(f'QCJob {job_id} completed')
+
+        '''
+        for project in self.project_data:
+            project_name = project['Sample_Project']
+            needs_human_filtering = project['HumanFiltering']
+
+            source_dir = join(self.output_path, project_name)
+
+
+            # TODO: UNCONFIRMED COMPLETED FILES WILL BE CREATED
+            # confirm Slurm job was successful by using .completed files.
+            # if not self._get_failed_indexes(project_name, job_id):
+            #     raise PipelineError("QCJob did not complete successfully.")
+
+            ### DANIEL'S NEW CODE DOES NOT APPEAR TO FILTER FOR
+                ZERO-LENGTH FILES ###
+
+            # determine where the filtered fastq files can be found and move
+            # the 'zero-length' files to another directory.
+            if needs_human_filtering is True:
+                filtered_directory = join(source_dir, 'filtered_sequences')
+            else:
+                filtered_directory = join(source_dir, 'trimmed_sequences')
+
+            empty_files_directory = join(source_dir, 'zero_files')
+            self._filter_empty_fastq_files(filtered_directory,
+                                           empty_files_directory,
+                                           self.minimum_bytes)
+        '''
+
+    def _process_sample_sheet(self):
+        sheet = KLSampleSheet(self.sample_sheet_path)
+        valid_sheet = validate_and_scrub_sample_sheet(sheet)
+
+        if not valid_sheet:
+            s = "Sample sheet %s is not valid." % self.sample_sheet_path
+            raise PipelineError(s)
+
+        header = valid_sheet.Header
+        chemistry = header['chemistry']
+
+        if header['Assay'] not in Pipeline.assay_types:
+            s = "Assay value '%s' is not recognized." % header['Assay']
+            raise PipelineError(s)
+
+        # Adapter trimming is not appropriate for metatranscriptomics, so it
+        # is not included here.
+        needs_adapter_trimming = header['Assay'] == Pipeline.METAGENOMIC_PTYPE
```

Comment:
Is this accurate?

Reply:
Two things are apparent: the first is that metatranscriptomics also requires adapter trimming and the second is that this is handled automatically by the new implementation as needed, based on the adapters in the fasta file. Thus, needs_adapter_trimming and self.needs_adapter_trimming is no longer needed. I've gone ahead and removed it.